tests/test_mesos_checkpointing.py

```diff
@@ -0,0 +1,124 @@
+        # are not running on the same host.
+        available_cpus = []
+        cluster_agents = sdk_cmd.cluster_request('GET', '/mesos/slaves').json()
+        for agent in cluster_agents["slaves"]:
+            available_cpus.append(int(float(agent["resources"]["cpus"])) - int(float(agent["used_resources"]["cpus"])))
+
+        available_cpus.sort(reverse=True)
+        assert len(available_cpus) >= 3, \
+            "Expected 3 or more nodes in the cluster to accommodate Dispatcher, " \
+            "Driver, and Executor each on a separate node"
+
+        global driver_cpus
+        driver_cpus = available_cpus[0]
+
+        global executor_cpus
+        executor_cpus = available_cpus[1]
+
+        log.info(f"{driver_cpus} cores will be used for driver, {executor_cpus} cores will be used for executor")
+        yield
+    finally:
+        utils.teardown_spark()
+
+
+@pytest.mark.sanity
+def test_agent_restart_with_checkpointing_disabled():
+    (driver_task, executor_task) = next(_submit_job_and_get_tasks())
+
+    driver_ip = sdk_networks.get_task_host(driver_task)
+    executor_ip = sdk_networks.get_task_host(driver_task)
+
+    # Dispatcher starts Driver Tasks with checkpointing enabled so Driver is expected to be in RUNNING state
+    _restart_task_agent_and_verify_state(driver_ip, driver_task, "TASK_RUNNING")
+    _restart_task_agent_and_verify_state(executor_ip, executor_task, "TASK_KILLED")
+
+
+@pytest.mark.sanity
+def test_agent_restart_with_checkpointing_enabled():
+    (driver_task, executor_task) = next(_submit_job_and_get_tasks(extra_args=["--conf spark.mesos.checkpoint=true"]))
+
+    driver_ip = sdk_networks.get_task_host(driver_task)
+    executor_ip = sdk_networks.get_task_host(driver_task)
+
+    _restart_task_agent_and_verify_state(executor_ip, executor_task, "TASK_RUNNING")
+    _restart_task_agent_and_verify_state(driver_ip, driver_task, "TASK_RUNNING")
+
+
+def _restart_task_agent_and_verify_state(host_ip, task, expected_state):
+    dcos_utils.agent_ssh(host_ip, stop_agent_cmd)
+    _check_agent_status(host_ip, "inactive")
+    dcos_utils.agent_ssh(host_ip, start_agent_cmd)
+    _check_agent_status(host_ip, "active")
+    _wait_for_task_status(task["id"], expected_state)
+
+
+@retrying.retry(
+        wait_fixed=5000,
+        stop_max_delay=120 * 1000,
+        retry_on_result=lambda res: not res)
+def _check_agent_status(host_ip, expected_status):
+    status = dcos_utils.agent_ssh(host_ip, check_agent_cmd)
+    log.info(f"Checking status of agent at host {host_ip}, expected: {expected_status}, actual: {status}")
+    return expected_status == status
+
+
+@retrying.retry(
+        wait_fixed=5000,
+        stop_max_delay=120 * 1000,
+        retry_on_result=lambda res: not res)
+def _wait_for_task_status(task_id, expected_state):
+    completed = expected_state != "TASK_RUNNING"
+    task = shakedown.get_task(task_id, completed=completed)
```

Comment:
can we avoid using `shakedown` in new code/tests?

Reply:
Unfortunately, there's no alternative in the current `testing` package and all the relevant tools there rely on `shakedown`. I expect the cleanup to happen in the scope of [DCOS-49680](https://jira.mesosphere.com/browse/DCOS-49680) otherwise this needs to be implemented from scratch. This looks like `agent_ssh` but with the only difference that `paramiko` dependency of `shakedown` was blocking the progress whereas `shakedown.get_task` is used across the whole codebase and works without issues.